iothub/device/src/DeviceAuthenticationWithX509Certificate.cs

```diff
@@ -40,6 +42,13 @@ public string DeviceId
         /// </summary>
         public X509Certificate2 Certificate { get; set; }
 
+        /// <summary>
+        /// Full chain of certificates from the one used to sign the device certificate to the one uploaded to the service. Private keys are not required for these certificates.
+        /// This is only supported on AMQP_Tcp_Only and Mqtt_Tcp_Only
```

Comment:
q - what was the restriction for ws? was it that the .net ws implementation wasn't forwarding the chain certificate info to the socket layer?

Reply:
No the socket layer never forwards it. It somehow does not work for ws. I did not dive deep as that code is out of our control and no customer was asking for it in particular.